mkdocs/docs/kubernetes-architecture.md

```diff
@@ -1,8 +1,105 @@
-# Kubernetes Architecture
+# Introduction to Kubernetes Architecture
+
+## Overview
+Kubernetes is a powerful container orchestration platform that automates deployment, scaling, and management of containerized applications. Its architecture is divided into two main parts:
+
+- Control Plane
+- Worker Nodes
+
+---
 
 ## Control Plane
-- etcd
 
-## Worker Node
-A worker node is a VM or physical computer that provides a worker in a Kubernetes system
-- Kubelet: communicates with the control plane
+The **Control Plane** is the brain of the Kubernetes cluster. It manages the cluster's state and makes decisions about scheduling, scaling, and responding to events. The Control Plane has individual components running as pods on the node, each responsible for various tasks. 
```

Comment:
Nice to have: a reference link on `pods` to a part of the docs where we explain what they are